frontend/src/components/Button.jsx

```diff
@@ -0,0 +1,55 @@
+import PropTypes from 'prop-types';
+
+export default function Button({
+  title = 'Title',
+  children,
+  component: Component = 'button',
+  variant = 'primary',
+  onClick,
+  style,
+  ...others
+}) {
+  title = children ?? title;
+  let buttonClasses = '';
+
+  switch (variant) {
+    case 'primary':
+      buttonClasses = 'button--primary';
+      break;
+    case 'success':
+      buttonClasses = 'button--success';
+      break;
+    case 'danger':
+      buttonClasses = 'button--danger';
+      break;
+    case 'warning':
+      buttonClasses = 'button--warning';
+      break;
+  }
```

Comment:
utilise plutôt un objet, les switch c'est de la merde

```suggestion
  const buttonClass = variantClasses[variant]
  
  const variantClasses = {
    primary: 'button--primary',
    success: 'button--success',
    danger: 'button--danger',
    warning: 'button--warning',
  }
```